### PR TITLE
use index version of calculating hash

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
panic hit on simulation with capitalization mismatch after hash calculation.
#### Summary of Changes
revert to previous hash calculation algorithm.
Fixes #

### CODE REVIEW

**Comment by jeffwashington** [comment]
#15180

**Comment by mvines** [comment]
(adding v1.5, as I think we want this patch there too)

File: runtime/src/accounts_background_service.rs [hunk 1]
```diff
@@ -122,8 +122,11 @@ impl SnapshotRequestHandler {
 
                 let mut hash_time = Measure::start("hash_time");
                 let mut hash_for_testing = None;
-                snapshot_root_bank
-                    .update_accounts_hash_with_index_option(true, test_hash_calculation);
+                const DO_NOT_USE_INDEX: bool = false;
+                snapshot_root_bank.update_accounts_hash_with_index_option(
+                    DO_NOT_USE_INDEX,
+                    test_hash_calculation,
+                );
```
Comment (sakridge) [inline runtime/src/accounts_background_service.rs:129]: unrelated, but we could write hash_for_testing as 
```
let hash_for_testing = if test_hash_calculations {
     Some(..)
} else {
     None
 };
 ```

**Review by sakridge — APPROVED** [review]

File: runtime/src/accounts_background_service.rs [hunk 2]
```diff
@@ -122,8 +122,11 @@ impl SnapshotRequestHandler {
 
                 let mut hash_time = Measure::start("hash_time");
                 let mut hash_for_testing = None;
-                snapshot_root_bank
-                    .update_accounts_hash_with_index_option(true, test_hash_calculation);
+                const DO_NOT_USE_INDEX: bool = false;
```
Comment (CriesofCarrots) [inline runtime/src/accounts_background_service.rs:125]: Fwiw, I find the double-negative a little difficult to follow. If the churn isn't too awful, could we update this to `use_index` in `update_accounts_hash_with_index_option` and etc?

**Comment by jeffwashington** [comment]
stable build failed once. I ran the test locally and it passed. I retry'd the build and it succeeded and is continuing.
https://buildkite.com/solana-labs/solana/builds/39884#_
test: test_stake_authorize_with_fee_payer

File: runtime/src/accounts_background_service.rs (hunk 1 above)
Comment (jeffwashington) [reply runtime/src/accounts_background_service.rs:129]: handled in #15191

File: runtime/src/accounts_background_service.rs [hunk 3]
```diff
@@ -122,8 +122,11 @@ impl SnapshotRequestHandler {
 
                 let mut hash_time = Measure::start("hash_time");
                 let mut hash_for_testing = None;
-                snapshot_root_bank
-                    .update_accounts_hash_with_index_option(true, test_hash_calculation);
+                const USE_INDEX: bool = true;
+                snapshot_root_bank.update_accounts_hash_with_index_option(
+                    !USE_INDEX,
```
Comment (CriesofCarrots) [inline runtime/src/accounts_background_service.rs:127]: I actually meant to continue the inversion upstream
https://github.com/solana-labs/solana/blob/88f22c360bce341a4aa10564f5f3ae76df8312c7/runtime/src/bank.rs#L4317, https://github.com/solana-labs/solana/blob/master/runtime/src/accounts_db.rs#L3890
So all `do_not_use_index` > `use_index`

But this works for now if that's too much churn for this PR
But

Comment (jeffwashington) [reply runtime/src/accounts_background_service.rs]: My preference would be to change this separately, once the dust settles, and it seems you are ok with that.

Comment (CriesofCarrots) [reply runtime/src/accounts_background_service.rs]: Yep, works for me.

**Review by CriesofCarrots — APPROVED** [review]

**Pull request merged**